### PR TITLE
New: Deprecated Notice control

### PR DESCRIPTION
**Pull request by nuritsha**

### CODE REVIEW

File: includes/base/widget-base.php [hunk 1]
```diff
@@ -774,4 +774,34 @@ public function remove_skin( $skin_id ) {
 	public function get_skins() {
 		return Plugin::$instance->skins_manager->get_skins( $this );
 	}
+
+	/**
+	 * @param string $plugin_title  Elementor-Pro | Elementor-Core
```
Comment (matipojo) [inline includes/base/widget-base.php]: Plugin title (not  "Elementor-Pro | Elementor-Core")

File: includes/base/widget-base.php [hunk 2]
```diff
@@ -774,4 +774,34 @@ public function remove_skin( $skin_id ) {
...
+	 * @param string $since         Plugin version widget was deprecated
+	 * @param string $last          Plugin version in which the widget will be removed
+	 * @param string $replacement   Widget replacement
+	 */
+	protected function deprecated_notice( $plugin_title, $since, $last, $replacement ) {
+		$this->start_controls_section( 'Deprecated',
+			[
+				'tab' => Controls_Manager::TAB_CONTENT,
```
Comment (matipojo) [inline includes/base/widget-base.php]: Remove. It's the default.

File: includes/base/widget-base.php [hunk 3]
```diff
@@ -774,4 +774,34 @@ public function remove_skin( $skin_id ) {
...
 		return Plugin::$instance->skins_manager->get_skins( $this );
 	}
+
+	/**
+	 * @param string $plugin_title  Elementor-Pro | Elementor-Core
+	 * @param string $since         Plugin version widget was deprecated
+	 * @param string $last          Plugin version in which the widget will be removed
+	 * @param string $replacement   Widget replacement
+	 */
+	protected function deprecated_notice( $plugin_title, $since, $last, $replacement ) {
```
Comment (matipojo) [inline includes/base/widget-base.php]: `$last, $replacement` should be optionally.

File: includes/base/widget-base.php [hunk 4]
```diff
@@ -774,4 +774,34 @@ public function remove_skin( $skin_id ) {
...
+				'tab' => Controls_Manager::TAB_CONTENT,
+				'label' => __( 'Deprecated', 'elementor' ),
+			]
+		);
+
+		$this->add_control(
+			'deprecated_notice',
+			[
+				'type' => Controls_Manager::DEPRECATED_NOTICE,
+				'tag' => $this->get_title(),
```
Comment (matipojo) [inline includes/base/widget-base.php]: Rename: tag => widget.

File: includes/editor.php [hunk 5]
```diff
@@ -667,6 +667,12 @@ public function enqueue_scripts() {
 				// Hotkeys screen
 				'keyboard_shortcuts' => __( 'Keyboard Shortcuts', 'elementor' ),
 
+				// Deprecated Control
+				'deprecated_notice' => __( 'The <strong>{{{ tag }}}</strong> widget has been deprecated since {{{ plugin }}} {{{ since }}}.
+											<br>It has been replaced by <strong>{{{ replacement }}}</strong>.
```
Comment (matipojo) [inline includes/editor.php]: Too match tabs?

File: includes/plugin.php [hunk 6]
```diff
@@ -611,6 +611,10 @@ private function __construct() {
 		add_action( 'init', [ $this, 'init' ], 0 );
 		add_action( 'rest_api_init', [ $this, 'on_rest_api_init' ] );
 	}
+
+	final public static function get_title() {
+		return 'Elementor';
```
Comment (matipojo) [inline includes/plugin.php]: Translate

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/plugin.php (hunk 6 above)
Comment (nuritsha) [reply includes/plugin.php]: done

File: includes/base/widget-base.php (hunk 4 above)
Comment (nuritsha) [reply includes/base/widget-base.php]: done

File: includes/base/widget-base.php (hunk 3 above)
Comment (nuritsha) [reply includes/base/widget-base.php]: done

File: includes/base/widget-base.php (hunk 2 above)
Comment (nuritsha) [reply includes/base/widget-base.php]: done

File: includes/base/widget-base.php (hunk 1 above)
Comment (nuritsha) [reply includes/base/widget-base.php]: done

File: includes/controls/deprecated-notice.php [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+ *
+ * A base control specific for creating Deprecation Notices control.
+ * Displays a warning notice in the panel.
+ *
+ * @since 2.6.0
+ */
+class Control_Deprecated_Notice extends Base_UI_Control {
+
+	/**
+	 * Get edprecated-notice control type.
```
Comment (matipojo) [inline includes/controls/deprecated-notice.php]: Typo: edprecated

File: includes/controls/deprecated-notice.php [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	 *
+	 * Retrieve the control type, in this case `raw_html`.
```
Comment (matipojo) [inline includes/controls/deprecated-notice.php]: raw_html??

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/controls/deprecated-notice.php (hunk 7 above)
Comment (nuritsha) [reply includes/controls/deprecated-notice.php]: done

File: includes/controls/deprecated-notice.php (hunk 8 above)
Comment (nuritsha) [reply includes/controls/deprecated-notice.php]: hmm... done

**Review by matipojo — APPROVED** [review]

**Pull request merged**